### PR TITLE
Added array comparison for aud parameter

### PR DESCRIPTION
**Pull request by thiagodiniz**
The aud parameter was tested only against a string, but if a project has more than one client id all the ids aren't returned at the json. So I added a option to pass an array os a string as a aud parameter.

### CODE REVIEW

**Comment by baxang** [comment]
Thanks @thiagodiniz! and bump @hxiong388 😃 It seems this is the most desired update as there are a few forks that implemented the same functionality. (I had to write some custom code to support multiple auds too) Can you take a look?

**Comment by hxiong388** [comment]
CC @dazuma

File: lib/google-id-token.rb [hunk 1]
```diff
@@ -132,7 +132,9 @@ def check_cached_certs(token, aud, cid)
       end
 
       if payload
-        if !(payload.has_key?('aud') && payload['aud'] == aud)
+        unless payload.has_key?('aud') &&
```
Comment (pusewicz) [inline lib/google-id-token.rb]: I'd suggest simplifying this code a bit: 

```suggestion
        if !(payload.has_key?('aud') && Array(payload['aud']).include?(aud))
```

This way we always operate on an array.

```ruby
>> Array(["abc"])
["abc"]
>> Array("abc")
["abc"]
```

**Comment by thiagodiniz** [comment]
@pusewicz @dazuma Pull-request updated!

**Comment by pusewicz** [comment]
@thiagodiniz I don't think you have pushed the changes!

**Comment by thiagodiniz** [comment]
@googlebot  I consent

**Comment by thiagodiniz** [comment]
@pusewicz can you consent to googlebot ?

File: lib/google-id-token.rb [hunk 2]
```diff
@@ -132,7 +132,9 @@ def check_cached_certs(token, aud, cid)
       end
 
       if payload
-        if !(payload.has_key?('aud') && payload['aud'] == aud)
+        if !(payload.has_key?('aud') && Array(payload['aud']).include?(aud))
+               (payload['aud'] == aud ||
+               (aud.is_a?(Array) && aud.include?(payload['aud'])))
```
Comment (pusewicz) [inline lib/google-id-token.rb]: This is still not right. The whole block should be:

```ruby
if !(payload.has_key?('aud') && Array(payload['aud']).include?(aud))
  raise AudienceMismatchError, 'Token audience mismatch'
end
```

**Comment by pusewicz** [comment]
@googlebot I consent.

File: lib/google-id-token.rb (hunk 2 above)
Comment (pusewicz) [reply lib/google-id-token.rb]: Sorry, it should be:

```ruby
if !(payload.has_key?('aud') && Array(aud).include?(payload['aud']))
  raise AudienceMismatchError, 'Token audience mismatch'
end
```

**Comment by thiagodiniz** [comment]
@googlebot I consent

**Comment by thiagodiniz** [comment]
@dazuma Any news?

**Comment by piotrj** [comment]
Well, seems that 2 years later this change is still hanging. Is that project dead on Google side? Is there any other ruby gem that does the same thing?

**Comment by thiagodiniz** [comment]
@piotrj  good question!

**Comment by baxang** [comment]
@dazuma Do you still manage this project? Or it would be nice if you are able to point someone to look after this PR.

**Comment by dazuma** [comment]
Sorry, this repo kept falling off the radar. I need to get it moved into the org for the Google client libraries teams so that it's visible to us. For now, this change looks fine.

**Comment by baxang** [comment]
Thank you very much @dazuma! It would great if you are able to release a new version of the gem, too!

**Comment by awcodify** [comment]
Merged when i already resigned 😅

**Comment by dazuma** [comment]
I'll get one pushed out in the next few days. We're evaluating this gem for some additional use cases, and need to make sure there aren't any issues.

**Comment by baxang** [comment]
That sounds fantastic! Appreciate @dazuma and stay safe 😄

**Comment by AlexMcConnell** [comment]
@dazuma Looks like that release didn't happen? Is there something else I should be using to validate requests from my flutter app?

**Comment by dazuma** [comment]
Oh yes, the logic from this gem was moved into the googleauth gem. See https://googleapis.dev/ruby/googleauth/latest/Google/Auth/IDTokens.html. The google-id-token gem is deprecated, and we should probably add some documentation to that effect.

**Pull request merged**